Testing/CICDTests/Services/SolutionServiceTests.cs

```diff
@@ -0,0 +1,310 @@
+﻿// <copyright file="SolutionServiceTests.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Abstractions;
+using System.Reflection;
+using CICDSystem;
+using CICDSystem.Services;
+using FluentAssertions;
+using Moq;
+using Nuke.Common.ProjectModel;
+using Xunit;
+using Project = Nuke.Common.ProjectModel.Project;
+
+namespace CICDSystemTests.Services;
+
+/// <summary>
+/// Tests the <see cref="SolutionService"/> class.
+/// </summary>
+[SuppressMessage("StyleCop.CSharp.OrderingRules",
+    "SA1202:Elements should be ordered by access",
+    Justification = "Not required for unit tests.")]
+public class SolutionServiceTests
+{
+    private readonly Mock<ISolutionWrapper> mockSolutionWrapper;
+    private readonly Mock<IFile> mockFile;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SolutionServiceTests"/> class.
+    /// </summary>
+    public SolutionServiceTests()
+    {
+        this.mockSolutionWrapper = new Mock<ISolutionWrapper>();
+        this.mockFile = new Mock<IFile>();
+    }
+
+    #region Constructor Tests
+    [Fact]
+    public void Ctor_WithNullSolutionWrapperParam_ThrowsException()
+    {
+        // Arrange & Act
+        var act = () =>
+        {
+            _ = new SolutionService(
+                null,
+                this.mockFile.Object);
+        };
+
+        // Assert
+        act.Should()
+            .Throw<ArgumentNullException>()
+            .WithMessage("The parameter must not be null. (Parameter 'solutionWrapper')");
+    }
+
+    [Fact]
+    public void Ctor_WithNullFileParam_ThrowsException()
+    {
+        // Arrange & Act
+        var act = () =>
+        {
+            _ = new SolutionService(
+                this.mockSolutionWrapper.Object,
+                null);
+        };
+
+        // Assert
+        act.Should()
+            .Throw<ArgumentNullException>()
+            .WithMessage("The parameter must not be null. (Parameter 'file')");
+    }
+    #endregion
+
+    #region Prop Tests
+    [Fact]
+    public void AllProjects_WhenGettingValue_ReturnsCorrectResult()
+    {
+        // Arrange
+        var project = CreateProjectObj("TestProject");
+        var expected = new ReadOnlyCollection<Project>(new[] { project });
+
+        this.mockSolutionWrapper.SetupGet(p => p.AllProjects)
+            .Returns(new ReadOnlyCollection<Project>(new[] { project }));
+
+        var sut = CreateService();
+
+        // Act
+        var actual = sut.AllProjects;
+
+        // Assert
+        actual.Should().BeEquivalentTo(expected);
+    }
+    #endregion
+
+    #region Method Tests
+    [Fact]
+    public void GetProject_WhenInvoked_ReturnsCorrectResult()
```

Comment:
I believe this method should be GetProjects.  That is what it's called above this block and below this block.
●
```suggestion
    public void GetProject_WhenInvoked_ReturnsCorrectResult()
```

Reply:
Nope,  this is correct.  I have 2 methods that get tested.  `GetProject()` and `GetProjects()`.
Look at lines 109 and 132.  These are the actual methods(functions) being executed for testing purposes.